packages/core/src/components/tabs/tabs.tsx

```diff
@@ -160,20 +232,22 @@ export class Tabs extends AbstractPureComponent<ITabsProps, ITabsState> {
     }
 
     public componentDidUpdate(prevProps: ITabsProps, prevState: ITabsState) {
-        if (this.state.selectedTabId !== prevState.selectedTabId) {
-            this.moveSelectionIndicator();
-        } else if (prevState.selectedTabId != null) {
-            // comparing React nodes is difficult to do with simple logic, so
-            // shallowly compare just their props as a workaround.
-            const didChildrenChange = !Utils.arraysEqual(
-                this.getTabChildrenProps(prevProps),
-                this.getTabChildrenProps(),
-                Utils.shallowCompareKeys,
-            );
-            if (didChildrenChange) {
+        window.requestAnimationFrame(() => {
```

Comment:
this should go at the lowest level, namely around `this.moveSelectionIndicator();`, not around the entire if blocks. especially since state could actually change between the function body and the animation frame.

Reply:
actually, it should probably go _inside_ `this.moveSelectionIndicator();` since we seem to do it every time now.